resolvers.ts

```diff
@@ -48,6 +48,98 @@ export const resolvers: Resolvers = {
         };
       }, []);
     },
+    bulkDownloads: async (root, args, context, info) => {
+      const statusDictionary = {
+        "success":"SUCCEEDED",
+        "error": "FAILED",
+        "waiting": "PENDING",
+        "running": "INPROGRESS",
+        //fyi: in NextGen there is also a status of STARTED
+      }
+      const urlParams = formatUrlParams({
+        searchBy: args?.input?.searchBy,
+        n:args?.input?.limit
+      })
+      const getEntityInputInfo = (entities) => {
+        return entities.map((entity) => {
+          return {
+            id: entity?.id,
+            name: entity?.sample_name,
+          }
+        })
+      };
+      const res = await get(`/bulk_downloads.json${urlParams}`, args, context);
+      const mappedRes = res.map(async (bulkDownload) => {
+        let url: string | null = null;
+        let entityInputs: {id: string, name: string}[] = [];
+        let sampleNames: Set<string> | null = null;
+        let totalSamples: number | null = null;
+        let description: string;
+        let file_type_display: string;
+        const details = await get(`/bulk_downloads/${bulkDownload?.id}.json`, args, context);
+        if (bulkDownload.status === "success"){
+          url = details?.bulk_download?.presigned_output_url;
+          entityInputs = [...getEntityInputInfo(details?.bulk_download?.workflow_runs), ...getEntityInputInfo(details?.bulk_download?.pipeline_runs)];
+          sampleNames = new Set(entityInputs.map((entityInput) => entityInput.name));
+          totalSamples = details?.bulk_download?.params?.sample_ids?.value?.length;
+        }
+        description = details?.download_type?.description;
+        file_type_display = details?.download_type?.file_type_display;
+
+        const { 
+          id, 
+          status, 
+          user_id, 
+          download_type,
+          created_at, 
+          download_name, 
+          output_file_size,
+          user_name, 
+          log_url,
+          analysis_type,
+          progress // --> to be discussed on Feb 16th, 2024
+          } = bulkDownload;
+
+          // In Next Gen we will have an array with all of the entity input 
+          // filtered through the nodes entity query to get the relevant info
+          // If there are 22 Consensus Genome Files coming from 20 Samples, there will be 42 items in the array.
+          // We will get `sampleNames` by checking __typename to see if the entity is a sample,
```

Comment:
is there a NextGen API that like returns a polymorphic type?

Reply:
I was thinking something like this:
`nodes(ids: "") {
    _id
    ... on ConsensusGenome {
      id
    }
    ... on GenomicRange {
      id
    }
    ... on Sample {
      id
      name
    }
  }`

Reply:
I see